internal/pkg/service/common/configmap/visit.go

```diff
@@ -15,6 +15,8 @@ type OnValue func(vc *VisitContext) error
 type AfterValue func(vc *VisitContext) error
 
 type VisitConfig struct {
+	// InitNilPtr initializes each found nil pointer with an empty struct.
+	InitNilPtr bool
```

Comment:
At the beginning, the `ConfigPatch` struct is empty, so all pointers are `nil`.

We need to set some nested values, so at first, we have to initialize `nil` pointers with empty structs.

We don't know, if a nested field will be set or not.

`AfterValue` callback (see next commit) ensures that empty struct are reverted - deleted at the end, after the all structure fields are binded. Only non-empty struct, with a nested value will remain.